### PR TITLE
new check added for testing production using double node or not(ocrvs-6918)

### PR DESCRIPTION
**Pull request by alsmk**
## Description

issue link : [#6918](https://github.com/opencrvs/opencrvs-core/issues/6918)
production environment must not use a single node cluster
new variable(single_node) introduced which will be false only for prod
and new check is added in ansible task

## Checklist

- [x] I have tested the changes locally, and written appropriate tests
- [x] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [x] I have updated the changelog with this change (if applicable)
- [x] I have updated the GitHub issue status accordingly

### CODE REVIEW

**Comment by alsmk** [comment]
If the production environment use single node cluster 
it will fail the provision like [this](https://github.com/opencrvs/opencrvs-farajaland/actions/runs/14309221428/job/40099996681)

File: infrastructure/server-setup/tasks/checks.yml [hunk 1]
```diff
@@ -16,3 +16,10 @@
       - elasticsearch_superuser_password is defined
       - disk_encryption_key is defined
       - encrypted_disk_size is defined
+
+- name: 'Check the production environment is using more than one node'
+  fail:
+    msg: 'The production environment must use more than one node. Please check the inventory file for production.'
```
Comment (Zangetsu101) [inline infrastructure/server-setup/tasks/checks.yml:22]:
```suggestion
    msg: 'The production environment should always use more than one node. Please check the inventory file for production.'
```

**Review by Zangetsu101 — APPROVED** [review]

File: infrastructure/server-setup/tasks/checks.yml [hunk 2]
```diff
@@ -16,3 +16,10 @@
...
+  when:
+    - groups['docker-workers'] | length == 0
```
Comment (rikukissa) [inline infrastructure/server-setup/tasks/checks.yml:24]: What happens if the group doesn't exist at all in the inventory rile?

**Comment by alsmk** [comment]
I have tested in my local infrastructure setup
I left docker-workers group empty 
```
docker-workers: {}
```
in that case the playbook fails like that:
```
TASK [Check the production environment is using more than one node] ************
task path: /home/dsi/Documents/opencrvs-countryconfig/infrastructure/server-setup/tasks/checks.yml:19
fatal: [manager]: FAILED! => {"changed": false, "msg": "The production environment should always use more than one node. Please check the inventory file for production."}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
manager                    : ok=3    changed=0    unreachable=0    failed=1    skipped=2    rescued=0    ignored=0   
```

File: infrastructure/server-setup/tasks/checks.yml (hunk 2 above)
Comment (alsmk) [reply infrastructure/server-setup/tasks/checks.yml:24]: in that case it is failing like this
```
TASK [Check the production environment is using more than one node] ************
task path: [/home/dsi/Documents/opencrvs-countryconfig/infrastructure/server-setup/tasks/checks.yml:19](https://file+.vscode-resource.vscode-cdn.net/home/dsi/Documents/opencrvs-countryconfig/infrastructure/server-setup/tasks/checks.yml:19)
fatal: [manager]: FAILED! => {"msg": "The conditional check 'groups['docker-workers'] | length == 0' failed. The error was: error while evaluating conditional (groups['docker-workers'] | length == 0): 'dict object' has no attribute 'docker-workers'\n\nThe error appears to be in '/home/dsi/Documents/opencrvs-countryconfig/infrastructure/server-setup/tasks/checks.yml': line 19, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: 'Check the production environment is using more than one node'\n  ^ here\n"}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
manager                    : ok=3    changed=0    unreachable=0    failed=1    skipped=2    rescued=0    ignored=0
```

Comment (alsmk) [reply infrastructure/server-setup/tasks/checks.yml:24]: should i add another condition for this in checks.yml ??

**Comment by alsmk** [comment]
When docker-workers doesn't exist at all , it will fail like that
```
TASK [Check the production environment is using more than one node] ************
task path: [/home/dsi/Documents/opencrvs-countryconfig/infrastructure/server-setup/tasks/checks.yml:19](https://file+.vscode-resource.vscode-cdn.net/home/dsi/Documents/opencrvs-countryconfig/infrastructure/server-setup/tasks/checks.yml:19)
fatal: [manager]: FAILED! => {"changed": false, "msg": "The production environment should always use more than one node. Please check the inventory file for production."}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
manager                    : ok=3    changed=0    unreachable=0    failed=1    skipped=2    rescued=0    ignored=0
```

File: infrastructure/server-setup/group_vars/all.yml [hunk 3]
```diff
@@ -14,3 +14,4 @@ backup_server_user: 'backup'
 backup_server_user_home: '/home/backup'
 crontab_user: root
 provisioning_user: provision
+single_node: true
```
Comment (adskyiproger) [inline infrastructure/server-setup/group_vars/all.yml:17]:
```suggestion
single_node: false
```

I guess we agreed to set this variable false by default.

That will be good compromise why we need explicitly mention variable per environment.

**Review by adskyiproger — APPROVED** [review]

File: infrastructure/server-setup/inventory/production.yml [hunk 4]
```diff
@@ -27,6 +27,7 @@ all:
       #- 55.55.55.55 # example jump server IP address
       []
     backup_server_remote_target_directory: /home/backup/backups
+    single_node: false
```
Comment (Zangetsu101) [inline infrastructure/server-setup/inventory/production.yml:30]: Let's leave a comment here mentioning why we are setting it to false

File: infrastructure/server-setup/inventory/production.yml [hunk 5]
```diff
@@ -27,6 +27,7 @@ all:
       #- 55.55.55.55 # example jump server IP address
       []
     backup_server_remote_target_directory: /home/backup/backups
+    single_node: false # Setting single_node to false implies that a single-node setup is not allowed in production.
```
Comment (Zangetsu101) [inline infrastructure/server-setup/inventory/production.yml:30]:
```suggestion
    single_node: false # At least 2 nodes are recommended for production environment. Set "single_node: true" at your own discretion.
```
